packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -86,6 +88,22 @@ class HotRunner extends ResidentRunner {
       // Already computed.
       return true;
     }
+
+    try {
+      // Will return immediately if pubspec.yaml is up-to-date.
+      await pubGet(
```

Comment:
> should this also add a new PubContext value that describes how the pub get is being called?

Ah, possibly, based on what the current message says? We may want it tweaked so it says slightly different things based on an initial flutter run vs. a restart.